packages/orn-locator/src/resolvers/books.ts

```diff
@@ -0,0 +1,155 @@
+import fetch from 'node-fetch';
+
+const oswebUrl = 'https://openstax.org/apps/cms/api/v2/pages';
+const fields = 'cnx_id,authors,publish_date,cover_color,amazon_link,book_state,promote_image,webview_rex_link,cover_url,title_image_url';
+
+const getArchiveInfo = async (bookId: string) => {
+  const releaseJson = await fetch('https://openstax.org/rex/release.json')
+    .then(response => response.json())
+  ;
+
+  const bookConfig = releaseJson.books[bookId];
+
+  return {
+    archivePath: bookConfig.archiveOverride || releaseJson.archiveUrl,
+    bookVersion: bookConfig.defaultVersion,
+  };
+};
+
+export const book = async(id: string) => {
+  const oswebData = await fetch(`${oswebUrl}?type=books.Book&fields=${fields}&cnx_id=${id}`)
+    .then(response => response.json() as any)
+    .then(data => data.items[0])
+  ;
+
+  return {
+    id,
+    orn: `https://openstax.org/orn/content:book/${id}`,
+    type: 'content:book' as const,
+    title: oswebData.title as string,
+    slug: oswebData.meta.slug as string,
+    theme: oswebData.cover_color as string,
+    urls: {
+      default: oswebData.meta.html_url as string,
+      information: oswebData.meta.html_url as string,
+      experience: oswebData.webview_rex_link as string,
+    },
+    images: {
+      default: oswebData.cover_url as string,
+      square: oswebData.cover_url as string,
+      wide: oswebData.title_image_url as string,
+      promotional: oswebData.promote_image.meta.download_url as string,
+    }
+  };
+};
+
+const mapTree = (bookId: string) => (tree: any) => {
+  if (tree.contents) {
+    const subTreeId = tree.id.split('@')[0];
+    return {
+      id: subTreeId as string,
+      title: tree.title as string,
+      contents: tree.contents.map(mapTree(bookId)),
+      orn: `https://openstax.org/orn/content:collection/${bookId}:${subTreeId}`,
```

Comment:
Do chapters already need an ORN entry?

Reply:
yea i'm going to use this to display the assignable content picker and its all organized by chapter and might allow assigning the chapter and treat that slightly differently than assigning all the individual sections